tools/HeII_input_file_maker.py

```diff
@@ -0,0 +1,280 @@
+        return J_E[0]
+
+    def dQ_hard_dz(self, redshift, E_lim = 1000.):
+        """Uniform heating rate from long MFP hard photons only (E_gamma > Emax), dQ/dz. This is making the assumption that all Helium is in the form of HeII."""
+        func = lambda E: ((E-self.E0_HeII)/E)*self.specific_intensity(redshift,E)*self.sigmaHeII(E)
+        w = scipy.integrate.quad(func,self.Emax,E_lim)
+        dQdz = 4.*np.pi*self.eVtoerg*self.cosmo.nHe(redshift)*w[0]*1./(self.cosmo.Hubble(redshift)*(1+redshift))
+        return dQdz
+
+    def dGamma_hard_dt(self, redshift, E_lim = 1000.):
+        """Photoionization heating of hard photons only (E_gamma > Emax), dGamma/dt. Units are erg/s/cm^3."""
+        func = lambda E: ((E-self.E0_HeII)/E)*self.specific_intensity(redshift,E)*self.sigmaHeII(E)
+        w = scipy.integrate.quad(func,self.Emax,E_lim)
+        dGammadt = 4.*np.pi*w[0]*self.eVtoerg*self.cosmo.nHe(redshift)
+        return dGammadt
+
+    def setUpInterpTable(self, outfile, numz = 100.):
+        """Built the interpolation table file, the main output of this code, loadable by the MP-Gadget reionization module."""
+        print("Setting up interpolation table!")
+
+        z_quasar = np.logspace(np.log10(6.0),np.log10(2.8),numz)
+        dQ_LMFP_dat = [self.dGamma_hard_dt(zqso) for zqso in z_quasar]
+        XHeIII = [self.hist.XHeIII(zqso) for zqso in z_quasar]
+
+        print('Creating table ',outfile)
+        with open(outfile , 'w') as f:
+            header = "#File parameters for this input file: Emax = %g, alpha_q = %g, Clumping factor = %g, Simple linear history or QSO history = %s\n" % (self.Emax, self.alpha_q, self.clumping_fac, self.hist)
+            f.write(header)
+            f.write('#Units of heating rate (3rd column) are erg/s/cm^3 \n')
+            f.write('{0:f} \n'.format(self.alpha_q))
+            f.write('{0:f} \n'.format(self.Emax))
+            for zqso, xHe, dQ_LMFP in zip(z_quasar, XHeIII, dQ_LMFP_dat):
+                f.write('{0:e} {1:e} {2:e} \n'.format(zqso, xHe, dQ_LMFP))
+
+class LinearHistory:
+    """Makes a HeII reionization history where X_HeIII is a linear function of redshift"""
+    def __init__(self, z_i, z_f):
+        self.z_i = z_i
+        self.z_f = z_f
+
+    def XHeIII(self, redshift):
+        """XHeIII history that is linear with redshift.
+           Set initial and final HeII reion redshifts (default is reion_z_f=2.8)."""
+        return np.clip((redshift - self.z_i) /(self.z_f-self.z_i), a_min = 0, a_max = 1)
+
+    def dXHeIIIdz(self, redshift):
+        """Change in the derivative of XHeIII, where XHeIII evolves linearly with redshift.
+           Set initial and final HeII reion redshifts (default is reion_z_f=2.8)."""
+        if self.z_f <= redshift <= self.z_i:
+            return 1./(self.z_f-self.z_i)
+        return 0.
+
+class QuasarHistory(LinearHistory):
+    """Determines the HeII reionization history from a quasar emissivity function"""
```

Comment:
QuasarHistory as subclass of a LinearHistory? 

They seem to be two different implementations of the HeIIReionHistory interface?

Reply:
LinearHistory is the simplest possible implementation of the reionization fraction, so I thought it made sense to have QuasarHistory extend it. The commonality is pretty small, so we could also make them separate classes.